### PR TITLE
Document BYOC

### PR DESCRIPTION
**Pull request by jordanstephens**
Changes in this PR

- [x] Create a "Deploy to your own cloud" tutorial
  - [x] Create a "Deploy to your own AWS Account" tutorial
  - [x] Create a "Deploy to your own Digital Ocean Account" tutorial
  - [x] Create a stub for "Deploy to your own GCP Account"
- [x] After getting started, "Deploy to your own cloud" is the next recommended tutorial
- [x] Document Deployment modes
  - [ ] create a "rolling updates" concept page
  - [ ] create a "databases" concept page
- [x] Add a few more items to the FAQ
- [x] Add more linkable subheadings to the AWS and Digital Ocean Provider pages

### CODE REVIEW

File: docs/use-cases/use-cases.md [hunk 1]
```diff
@@ -34,4 +34,4 @@ Some applications are designed to run in stateful environments. For example, a C
 
 ## Data Stores
 
-Defang is not a good fit for data stores like databases, caches, and message queues. These types of services are designed to be long-lived and to store data. Defang is primarily designed for stateless workloads, and it's not a good fit for stateful workloads: if a container is replaced, all the data in it is lost. You could probably coerce a data store to work in a stateless environment with certain replication strategies, but it's not a good fit.
\ No newline at end of file
+Defang is not a good fit for data stores like databases, caches, and message queues. These types of services are designed to be long-lived and to store data. Defang is primarily designed for stateless workloads, and it's not a good fit for stateful workloads: if a container is replaced, all the data in it is lost. You could probably coerce a data store to work in a stateless environment with certain replication strategies, but it's not a good fit.
```
Comment (Prakash-Sundaresan) [inline docs/use-cases/use-cases.md:37]: This is not accurate. We should mention the issue with stateful *services* and then point to Defang's support for managed storage options - https://docs.defang.io/docs/concepts/managed-storage

**Review by Prakash-Sundaresan — COMMENTED** [review]
See detailed comments.
Also, we should replace all references to "AI assistant" with "AI agent" - e.g. in https://docs.defang.io/docs/intro

File: docs/faq.md [hunk 2]
```diff
@@ -23,13 +23,37 @@ description: Frequently asked questions about Defang.
...
+
+### Can I deploy a service that depends on another service?
+
+- Defang does not currently support service dependencies. All services will be deployed simultaneously. Defang will however run multiple healthchecks before marking a service as healthy and spinning down any previously deployed services when using the `production` deployment mode.
+
+### Do you plan to support other clouds?
 
-- The [Defang Playground](./concepts/defang-playground.md) is meant for testing and trial purposes only. Deployment of productions apps with [Defang BYOC](./concepts/defang-byoc.md) is not yet supported and disallowed by the [Terms of Service](https://defang.io/terms-service.html). If you are interested in running production apps, please [contact us](https://defang.io/#Contact-us).
+- While we currently support [AWS](./concepts/defang-byoc#aws) and [Digital Ocean](./concepts/defang-byoc#digitalocean), we plan to support other clouds in future releases, such [GCP](./providers/gcp.md) and [Azure](./providers/azure.md).
```
[the remark this answers is not in the document]
Comment (commit111) [reply docs/faq.md]: fixed those typos

File: docs/tutorials/_category_.json [hunk 3]
```diff
@@ -1,9 +1,8 @@
 {
   "label": "Tutorials",
-  "position": 400,
+  "position": 300,
   "link": {
     "type": "generated-index",
     "description": "Check out these tutorials to learn more about the Defang platform."
```
[the remark this answers is not in the document]
Comment (jordanstephens) [reply docs/tutorials/_category_.json:6]: How about
```suggestion
    "description": "Check out these tutorials to learn more about how to use Defang."
```

[the remark this answers is not in the document]
Comment (raphaeltm) [reply docs/tutorials/_category_.json:6]: I like @jordanstephens 's suggestion. Just "Defang" without any additional wording.

File: docs/use-cases/use-cases.md (hunk 1 above)
Comment (raphaeltm) [reply docs/use-cases/use-cases.md:37]: @Prakash-Sundaresan I think you're right. This is no longer accurate. I do think we should have a place where we caution people against trying to run unsupported data stores until we support volume mounts, though. Like, at the moment people shouldn't be using Defang to run Mongo, or ElasticSearch or anything like that.

File: docs/concepts/deployment-modes.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+* **Development**: This mode is used for development and testing purposes. It typically involves less stringent resource allocations and may include debugging tools and verbose logging to aid in development.
+* **Staging**: This mode serves as a pre-production environment where applications are tested in conditions that closely mimic production. It helps in identifying issues that might not be apparent in the development environment.
+* **Production**: This mode is used for live deployments. It involves optimized configurations for performance, security, and reliability. Resource allocations are typically higher, and debugging tools are minimized to ensure stability.
+
+## Deployment Mode Comparison
+
+| Feature | Development | Staging | Production |
+|-|-|-|-|
+| Build Resources | Builds will be run with 2x vCPUs | Builds will be run with 2x vCPUs | Builds will be run with 4x vCPUs |
+| Databases | Defang will provision resources optimized for burstable memory | | Defang will provision resources optimized for production |
```
[the remark this answers is not in the document]
Comment (jordanstephens) [reply docs/concepts/deployment-modes.md:20]: Yes. I think we should create a `Databases` concept topic where we can discuss the subtlety more carefully 👍

**Review by lionello — APPROVED** [review]

**Pull request merged**